### PR TITLE
Graph Data Pills

### PR DESCRIPTION
**Pull request by Waidhoferj**
**Add a new pill type `GraphPill` to match the latest pills from the Figma Prototype.**
Screen Shot 2021-07-28 at 2 56 37 PM

## Changes

- [x] Add filters to pill
- [x] Add aggregations to pill
- [x] Add pill header
- [x] Handle events associated with pill
- [x] Rewrite axis-click logic to target new pill structure for selection

### CODE REVIEW

**Comment by Waidhoferj** [comment]
## Todos
- When the encoding is clicked, open up the same encoding choice drawer as the add button, but just assign the new encoding to the object of the old one and delete the old encoding.
- When the field is clicked, open up the variable selection menu. On item click, find the encoding with that field and replace the field name with the new one.

**Comment by Waidhoferj** [comment]
# Testing
- Verify that you can change encoding and field
- Verify that pills can be deleted
- Add a pill encoding, click on the empty tile and pick a field
- Verify that filter labels are added to pill when applied on filter screen
- Verify that 2 encodings with the same field get both of their filters updated when one is updated.

**Review by jahn96 — CHANGES_REQUESTED** [review]
Categorical variable shouldn't have aggregate icon.
I think the UI is a little different from the pill snap shot above. Is this going to be in another PR?

**Comment by Waidhoferj** [comment]
Ah yeah I missed the `oneOf` checkbox icon for the categorical pill as well. Thanks for flagging this.

File: src/components/Sidebar/Tabs/VariablesTab.tsx [hunk 1]
```diff
@@ -65,56 +85,113 @@ export default function VariablesTab({
   useEffect(() => {
     setActiveEncoding(clickedAxis);
   }, [clickedAxis]);
+  const saveSpecToHistory = useSpecHistory();
+  // A list of GraphPill Props. Defined separately from spec to prevent reordering during user edits.
+  const [pillsInfo, setPillsInfo] = useState<PillState[]>([]);
```
Comment (jahn96) [inline src/components/Sidebar/Tabs/VariablesTab.tsx:90]: I think this is great!

**Review by jahn96 — APPROVED** [review]
LGTM!

**Pull request merged**